packages/tab/index.tsx

```diff
@@ -79,34 +114,41 @@ export default class Tab extends Component {
 
   get adapter() {
     return {
-      addClass: (className) => {
+      addClass: (className: string) => {
         const classList = new Set(this.state.classList);
         classList.add(className);
         this.setState({classList});
       },
-      removeClass: (className) => {
+      removeClass: (className: string) => {
         const classList = new Set(this.state.classList);
         classList.delete(className);
         this.setState({classList});
       },
-      hasClass: (className) => this.classes.split(' ').includes(className),
-      setAttr: (attr, value) => this.setState({[attr]: value}),
-      getOffsetLeft: () => this.tabElement_.current && this.tabElement_.current.offsetLeft,
-      getOffsetWidth: () => this.tabElement_.current && this.tabElement_.current.offsetWidth,
-      getContentOffsetLeft: () => this.tabContentElement_.current && this.tabContentElement_.current.offsetLeft,
-      getContentOffsetWidth: () => this.tabContentElement_.current && this.tabContentElement_.current.offsetWidth,
+      hasClass: (className: string) => this.classes.split(' ').includes(className),
```

Comment:
It's unclear what your canonical data model is.  You have both `classList` and a getter `classes`.  `classList` contains some, but not all, of your classes.  You're delegating to an external library to generate a string, and then splitting it to see the results.

I don't know when/how often `hasClass` is called, but this is a surprising amount of logic for something that looks fairly simple.

Reply:
This is how we wrap the react components using the MDC Foundation/Adapter pattern. hasClass and any other adapter methods are used in MDC Web's foundation. 

We had to design for MDC's Foundation, the developer and static classes, hence the complexity.